From 76dc81c4cdeed20cd07c54ef1953284d7f029673 Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Thu, 27 Sep 2018 18:07:34 +0200
Subject: [PATCH 1/1] device: fix crash in nm_device_generate_connection()

Fixes: 89d1c9fb302e84b92d0be415f0bb4bdca34ae74c

https://bugzilla.redhat.com/show_bug.cgi?id=1631741
(cherry picked from commit f744e29dd368491adadf74976fc4d9e0bdbba32e)
(cherry picked from commit 3fc592219e106fdbf98e9e994a4531d9d045d0da)
---
 src/devices/nm-device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/devices/nm-device.c b/src/devices/nm-device.c
index 26ea06bda..f2dbe6594 100644
--- a/src/devices/nm-device.c
+++ b/src/devices/nm-device.c
@@ -5486,7 +5486,7 @@ nm_device_generate_connection (NMDevice *self,
 
 	klass->update_connection (self, connection);
 
-	if (!nm_connection_normalize (connection, NULL, NULL, error)) {
+	if (!nm_connection_normalize (connection, NULL, NULL, &local)) {
 		g_set_error (error, NM_DEVICE_ERROR, NM_DEVICE_ERROR_FAILED,
 		             "generated connection does not verify: %s",
 		             local->message);
-- 
2.17.1

